File: lib/src/features/auth/data/models/login_request.dart
class LoginRequest {
  const LoginRequest({
    required this.identifier,
    required this.password,
  });

  final String identifier;
  final String password;

  Map<String, Object?> toJson() {
    return <String, Object?>{
      'username': identifier.trim(),
      'password': password,
    };
  }
}

File: backend/src/main/java/com/myfschool/auth/dto/LoginRequest.java
package com.myfschool.auth.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record LoginRequest(
        @NotBlank(message = "Vui lòng nhập số điện thoại hoặc tên đăng nhập")
        @Size(max = 100, message = "Tên đăng nhập không được vượt quá 100 ký tự")
        String username,

        @NotBlank(message = "Vui lòng nhập mật khẩu")
        @Size(max = 128, message = "Mật khẩu không được vượt quá 128 ký tự")
        String password
) {
}
